### ISSUE TITLE
FIR data labels

### ISSUE THREAD

**Issue by connorlee77** [princeton-computational-imaging/SeeingThroughFog#12]
Hi, I was wondering if there are labels for the FIR data. I haven't found any in the folders. We've tried overlaying the RGB labels after normalizing them to the FIR image size. However, it seems like there is an offset, perhaps due to the placement of the FIR cameras. 

How did you guys handle this in the paper?

**Comment by bjajoh** [comment]
Did you find a solution to the thermal labels? @connorlee77

**Comment by connorlee77** [comment]
No, I haven't.

**Comment by manonthegithub** [comment]
+ 1. Looking for a solution of getting FIR labels.

**Comment by bjajoh** [comment]
One basic approach could be to just warp the fir images into the rgb plane and use rgb labels. It is not nice, but the best low effort solution.

**Comment by bjajoh** [comment]
I tested that approach and it's not useful at all. It's really sad that there're no public FIR labels..

**Comment by connorlee77** [comment]
@bjajoh how did you warp the fir images to RGB plane btw (I'm looking to do the same)? Yeah, I believe part of the problem is that RGB ground truth does not necessarily have to be ground truth for IR so training on RGB labels messes things up. 

Have you tried pretraining on FLIR thermal data and finetuning on your warped IR data? I'm thinking that would yield better results.

**Comment by bjajoh** [comment]
@connorlee77 I simply estimated the transformation matrix an applied it to all images. But apparently fir and rgb are so far appart, that it doesn't work geometrically on a good enough level. A lot of labels will just be next to the actual object if they're way closer or further then the calibrated transformation matrix. Exactly this is another problem, there are a lot of cases where the objects are only visible in one of both domains. Pretraining on other FIR data helps a lot to increase the accuracy (FLIR and KAIST datasets are good for this).

**Comment by connorlee77** [comment]
@bjajoh Gotcha, did you estimate the transformation matrix by handpicking matching features and computing a perspective transformation?

**Comment by bjajoh** [comment]
I tried all different approaches but the main issues is the geometrical displacement of the rgb and fir camera. Any scene that has some kind of depth (wich is basically everything) fails with this approach. I guess labeling all images by hand is the only real solution.

**Comment by MarioBijelic** [comment]
Hi guys, 

we did not include the FIR camera in the labelling process. But if you are able to provide labels for our dataset we would include them on our dataset page and list you as a contributor. Sorry I can not provide any better help. 

Best regards,
Mario
